Microsoft.Toolkit.Uwp.UI.Controls/Carousel/Carousel.cs

```diff
@@ -0,0 +1,387 @@
+                    {
+                        if (this.SelectedIndex > 0)
+                        {
+                            this.SelectedIndex--;
+                        }
+
+                        e.Handled = true;
+                    }
+
+                    break;
+                case Windows.System.VirtualKey.Left:
+                case Windows.System.VirtualKey.GamepadDPadLeft:
+                case Windows.System.VirtualKey.GamepadLeftThumbstickLeft:
+                    if (this.Orientation == Orientation.Horizontal)
+                    {
+                        if (this.SelectedIndex > 0)
+                        {
+                            this.SelectedIndex--;
+                        }
+
+                        e.Handled = true;
+                    }
+
+                    break;
+                case Windows.System.VirtualKey.Right:
+                case Windows.System.VirtualKey.GamepadDPadRight:
+                case Windows.System.VirtualKey.GamepadLeftThumbstickRight:
+                    if (this.Orientation == Orientation.Horizontal)
+                    {
+                        if (this.SelectedIndex < this.Items.Count - 1)
+                        {
+                            this.SelectedIndex++;
+                        }
+
+                        e.Handled = true;
+                    }
+
+                    break;
+            }
+        }
+
+        protected override void OnManipulationDelta(ManipulationDeltaRoutedEventArgs e)
+        {
+            if (carouselPanel == null)
+            {
+                carouselPanel = this.GetItemsPanel();
+            }
+
+            if (carouselPanel == null)
+            {
+                return;
+            }
+
+            carouselPanel.OnManipulationDelta(this, e);
+        }
+
+        protected override void OnManipulationCompleted(ManipulationCompletedRoutedEventArgs e)
+        {
+            if (carouselPanel == null)
+            {
+                carouselPanel = this.GetItemsPanel();
+            }
+
+            if (carouselPanel == null)
+            {
+                return;
+            }
+
+            carouselPanel.OnManipulationCompleted(this, e);
+        }
+
+        internal void OnPointerWheelChanged(object sender, PointerRoutedEventArgs e)
+        {
+            var index = e.GetCurrentPoint(null).Properties.MouseWheelDelta > 0 ? -1 : 1;
+            if (index == -1 && this.SelectedIndex > 0)
+            {
+                this.SelectedIndex--;
+            }
+
+            if (index == 1 && this.SelectedIndex < this.Items.Count - 1)
+            {
+                this.SelectedIndex++;
+            }
+
+            e.Handled = true;
+        }
+
+        protected override void PrepareContainerForItemOverride(DependencyObject element, object item)
+        {
+            base.PrepareContainerForItemOverride(element, item);
+
+            if (this.ItemTemplate == null)
```

Comment:
Oh Yes, my bad you're right on this point.
Anyway, adding a CarouselItem implies adding a new control class and a new style.
I want to have something the most simpliest.
Adding a CarouselItem is, in my point of view, a good thing if we need events and so on on each items.